modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/validation/ParameterTypeModelValidator.java

```diff
@@ -132,6 +146,17 @@ private void checkInvalidFieldAnnotations(NamedObject model, String typeName, Fi
           }
         }
       }
+
+      private boolean isPojoWithDefaultConstructor(Class<?> clazz) {
+        return (!clazz.isInterface() && !isAbstract(clazz.getModifiers())
+            && Stream.of(clazz.getConstructors())
+                .noneMatch(c -> c.getParameterCount() == 0));
+      }
+
+      private Class<?> getTypedValueGenericClass(Type type) {
+        return type.getGenerics().size() != 0 ? type.getGenerics().get(0).getConcreteType().getDeclaringClass().get()
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Use isEmpty() to check whether the collection is empty or not. ([squid:S1155](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS1155))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7615&issues=AWhiR_G7otIXnmsUW4nN&open=AWhiR_G7otIXnmsUW4nN)

[ISSUE_KEY]: AWhiR_G7otIXnmsUW4nN